dbcon/joblist/distributedenginecomm.cpp

```diff
@@ -386,9 +387,22 @@ void DistributedEngineComm::Listen(boost::shared_ptr<MessageQueueClient> client,
 Error:
     // @bug 488 - error condition! push 0 length bs to messagequeuemap and
     // eventually let jobstep error out.
-    /*	boost::mutex::scoped_lock lk(fMlock);
-    	//cout << "WARNING: DEC READ 0 LENGTH BS FROM " << client->otherEnd()<< endl;
-
+    if (fIsExeMgr)
+    {
+        boost::mutex::scoped_lock lk(fMlock);
+        std::cout << "WARNING: DEC READ 0 LENGTH BS FROM "
+            << client->otherEnd()<< " OR GOT AN EXCEPTION READING" << std::endl;
+        decltype(pmCount) originalPMCount = pmCount;
+        // Re-establish if a remote PM restarted.
+        std::this_thread::sleep_for(std::chrono::seconds(3));
+        Setup();
+        if (originalPMCount != pmCount)
+        {
+            ostringstream os;
+            os << "DEC: lost connection to " << client->addr2String();
+            writeToLog(__FILE__, __LINE__, os.str(), LOG_TYPE_ERROR);
+        }
+/*
     	MessageQueueMap::iterator map_tok;
```

Comment:
side note, this may be one reason why aborts have been a problem since I got here.  If DEC is no longer communicating PM connection failures to the jobsteps, well, IDK what else has been changed around this.  I hope there's something that tells the jobsteps to stop.

Reply:
Related Zhi code has been commented out since the initial commit so I can't answer you. I will test MCS bahavior removing PM when ExeMgr avaits PM results doing SELECT.

Reply:
I've restored this 0 message to notify joblist it is done in case of DEC errors in DEC::Listen() and ::writeToClient().